crates/driver/src/api/solve/dto/auction.rs

```diff
@@ -0,0 +1,137 @@
+use {
+    crate::{logic, util::serialize},
+    ethereum_types::{H160, U256},
+    num::BigUint,
+    serde::Deserialize,
+    serde_with::{serde_as, DisplayFromStr},
+    std::collections::HashMap,
+};
+
+impl From<Auction> for logic::competition::Auction {
+    fn from(_: Auction) -> Self {
+        todo!()
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Deserialize)]
+pub struct Auction {
+    id: u64,
+    block: u64,
+    orders: Vec<Order>,
+    deadline: chrono::DateTime<chrono::Utc>,
+    #[serde_as(as = "HashMap<_, DisplayFromStr>")]
+    prices: HashMap<H160, BigUint>,
+}
+
+// TODO Should we rename kind to side and class to kind in this interface?
+#[serde_as]
+#[derive(Debug, Deserialize)]
+struct Order {
+    #[serde_as(as = "serialize::Hex")]
+    id: Vec<u8>,
```

Comment:
naming-nit: I would call this `uid` - as it is a unique identifier for the order (two orders have the same UID if and only if they are they represent the same on-chain order) and because that is the domain-specific term we use elsewhere in the protocol.

Reply:
> IIRC auction ids are not unique?

Correct, they can collide across deployments.